File: mobile-wallet-master/core/src/main/java/org/mifos/mobilewallet/core/data/fineract/api/services/AuthenticationService.java
package org.mifos.mobilewallet.core.data.fineract.api.services;

import org.mifos.mobilewallet.core.data.fineract.api.ApiEndPoints;
import org.mifos.mobilewallet.core.data.fineract.entity.UserEntity;

import retrofit2.http.POST;
import retrofit2.http.Query;
import rx.Observable;

/**
 * Created by naman on 17/6/17.
 */

public interface AuthenticationService {

    @POST(ApiEndPoints.AUTHENTICATION)
    Observable<UserEntity> authenticate(@Query("username") String username,
            @Query("password") String password);

}

File: core/src/main/java/org/mifos/mobilewallet/core/data/fineractcn/api/services/AuthenticationService.kt
package org.mifos.mobilewallet.core.data.fineractcn.api.services

import org.mifos.mobilewallet.core.data.fineractcn.api.ApiEndPoints
import org.mifos.mobilewallet.core.data.fineractcn.entity.LoginResponse
import retrofit2.http.POST
import retrofit2.http.Query
import rx.Observable

/**
 * Created by Devansh 17/06/2020
 */
interface AuthenticationService {

    @POST(ApiEndPoints.IDENTITY + "/token")
    fun authenticate(@Query("grant_type") grantType: String,
                     @Query("username") userName: String,
                     @Query("password") password: String): Observable<LoginResponse>

}